homeassistant/components/suez_water/coordinator.py

```diff
@@ -48,61 +34,38 @@ def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
         )
 
     async def _async_setup(self) -> None:
-        self._sync_client = await self.hass.async_add_executor_job(self._get_client)
+        self._async_client = await self._get_client()
+        self._data_api = SuezData(self._async_client)
 
     async def _async_update_data(self) -> AggregatedSensorData:
         """Fetch data from API endpoint."""
         async with asyncio.timeout(30):
-            return await self.hass.async_add_executor_job(self._fetch_data)
+            try:
+                data = await self._fetch_data()
```

Comment:
Let's inline the function here

Reply:
Yeah when applying all you other comments there should be much left.
But without those I personally find it cleaner to split data fetching in it's own method.

Reply:
I mostly find it strange that we have 2 try blocks, one for catching an exception and one for closing a session, while they could be merged and then its still only a single line of code :)

Reply:
Yes I understand that part.

I wont hide I did split it originally when I had all the changes on the same PR 😄